src/google/protobuf/compiler/importer.cc

```diff
@@ -290,11 +290,11 @@ static std::string CanonicalizePath(std::string path) {
   std::vector<std::string> canonical_parts;
   std::vector<std::string> parts = Split(
       path, "/", true);  // Note:  Removes empty parts.
-  for (int i = 0; i < parts.size(); i++) {
-    if (parts[i] == ".") {
+  for (auto && part : parts) {
```

Comment:
I think most (or all) of these don't need to be rvalue references since they're not being modified, so for example in this case it should be `const std::string&` or `const auto&` instead of `auto &&`.

Reply:
Thanks, done